solver/utils.go

```diff
@@ -14,6 +19,45 @@ var (
 	differ = gojsondiff.New()
 )
 
+func getDocumentDiff(a, b *state.Document) (string, error) {
+	aCopy := a.ShallowCopy()
+	bCopy := a.ShallowCopy()
+	aContent := *a.Content
+	bContent := *b.Content
+	aCopy.Content = nil
+	bCopy.Content = nil
+	objDiff, err := getDiff(aCopy, bCopy)
+	if err != nil {
+		return "", err
+	}
+	var contentDiff string
+	if json.Valid([]byte(aContent)) {
+		var aObj, bObj interface{}
+		err = json.Unmarshal([]byte(aContent), &aObj)
+		if err != nil {
+			return "", err
+		}
+		err = json.Unmarshal([]byte(bContent), &bObj)
+		if err != nil {
+			return "", err
+		}
+		aBytes, err := json.MarshalIndent(aObj, "", "\t")
+		if err != nil {
+			return "", err
+		}
+		bBytes, err := json.MarshalIndent(bObj, "", "\t")
+		if err != nil {
+			return "", err
+		}
+		aContent = string(aBytes)
+		bContent = string(bBytes)
+	}
+	edits := myers.ComputeEdits(span.URIFromPath("old"), aContent, bContent)
+	contentDiff = fmt.Sprint(gotextdiff.ToUnified("old", "new", aContent, edits))
```

Comment:
Can we please rename `old` and `new` to actual path to the file?

Reply:
Both versions share the same local path--sort of, anyway, since one was pulled from the Konnect API. There aren't actually separate local filepaths to show.

We potentially could reconstruct the local paths and then append suffixes to get something like `servicePackage/serviceVersion/foo.json.old` and `servicePackage/serviceVersion/foo.json.new`, but it's rather convoluted, since these are state.Document structs (which contain the path value we actually get from/push to Konnect) rather than file.Document structs (where the path is the local path, which you see in konnect.yaml). IMO trying to pass that along from the konnect.yaml read onward into the state struct and hiding it from the diff mechanism probably isn't worth the effort at present.